Add lesson plan for Collaborating with GitHub session Feb 22

sjeblee opened the pull request.

sjeblee committed:
Added lesson plan for github collaboration

sjeblee committed:
Added lesson plan for github collaboration

sjeblee committed:
Update lesson plan for github collaboration

sjeblee committed:
Fix a typo in github collaboration lession

sjeblee committed:
Merge branch 'gh-pages' of https://github.com/sjeblee/studyGroup into gh-pages

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+---
+layout: page
+title:
+visible: false
```
Set to `true`.

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+<!-- change visible to true if you want it on the site -->
+<!-- remove any tags listed above that are not relevant -->
+
+# Collaborating on GitHub
```
Put this in `title:` above and remove this line instead.

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+ - **Authors**: Serena Jeblee
+ - **Research field**: Computational Linguistics
+ - **Lesson topic**: Collaborating on GitHub
+ - **Lesson content URL**: <link>
```
https://github.com/UofTCoders/studyGroup/tree/lessons/git/collaboration/

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+- Branches
+- Rebasing
+
+## Whirlwind Overview of Github
```
GitHub

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+- What is version control?
+    - version control helps you collaborate on code especially when your changes might overlap
+    - master repo vs local repos
+    - master - shared version stored on github or a server
```
GitHub

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+- How to contribute code
+~~~
+git clone [repo-url]
+git pull // syncs latest version from the master repo to your local repo
```
`#` for shell comments.

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+    - make sure the commit message is descriptive
+
+## Code commit process
+- Create a new repo on github
```
GitHub

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+~~~
+- Unstage the changes, but keep them
+~~~
+    git reset --hard HEAD and git reset HEAD~x
```
Split lines; `and` doesn't mean anything in shell.

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -0,0 +1,120 @@
+- Add the junk code back and commit it
+- Undo a local commit
+~~~
+    git reset --soft HEAD~1 \\ keeps changes, but undoes the commit
```
Slashes are backward, but use `#` instead.

QuLogic reviewed (commented):
I'm not sure how others feel, but I think this could use a smidge more GitHub and a tiny bit less git review. What to do with PRs, commenting, merging, reviewing, etc.

joelostblom commented:
I agree that it would be great with more information around PRs, including how to review them and go back and forth with comments and changes to the PR before merging it!

sjeblee committed:
Update git collaboration lesson

Fix errors and address comments
Add more info about forking, pull requests, and code reviews

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -11,103 +11,80 @@ tags:
  - **Research field**: Computational Linguistics
  - **Lesson topic**: Collaborating on GitHub
- - **Lesson content URL**: <link>
+ - **Lesson content URL**: https://github.com/UofTCoders/studyGroup/tree/lessons/git/collaboration/
```
The `<>` are still needed to make this a link.

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -118,3 +95,31 @@ git commit
+## Branches
+- Branches allow you to have different commit histories in the same repo
+- Useful for feature work
+- For more info see: https://github.com/UofTCoders/studyGroup/blob/gh-pages/lessons/git/branches/lesson.md
```
Also add `<>` around this link.

QuLogic commented on lessons/git/collaboration/lesson.md:
```diff
@@ -118,3 +95,31 @@ git commit
+## Merging the Pull Request
+- Requires push access to the repo
+- If there are conflicts, you'll have to check out the pull request and resolve them, otherwise you can merge online
+- For more info on how to merge a pull request, see https://help.github.com/articles/merging-a-pull-request/
```
`<>` here too.

QuLogic reviewed (commented):
Looks good; hope we have time for a demo/hands-on of the PR stuff.

sjeblee commented:
Yes, the plan is to do a hands-on exercise with pull requests.

sjeblee committed:
Fix links in the GitHub Collaboration lesson
